src/chttpd/src/chttpd_misc.erl

```diff
@@ -114,36 +119,48 @@ handle_all_dbs_req(#httpd{method='GET'}=Req) ->
     Etag = couch_httpd:make_etag({Info}),
     Options = [{user_ctx, Req#httpd.user_ctx}],
     {ok, Resp} = chttpd:etag_respond(Req, Etag, fun() ->
-        {ok, Resp} = chttpd:start_delayed_json_response(Req, 200, [{"ETag",Etag}]),
+        {ok,Resp}=chttpd:start_delayed_json_response(Req,200,[{"ETag",Etag}]),
         VAcc = #vacc{req=Req,resp=Resp},
-        fabric:all_docs(ShardDbName, Options, fun all_dbs_callback/2, VAcc, Args)
+        fabric:all_docs(ShardDbName, Options,
+            fun all_dbs_or_dbs_info_callback/2, VAcc, Args)
     end),
     case is_record(Resp, vacc) of
         true -> {ok, Resp#vacc.resp};
         _ -> {ok, Resp}
-    end;
-handle_all_dbs_req(Req) ->
-    send_method_not_allowed(Req, "GET,HEAD").
+    end.
 
-all_dbs_callback({meta, _Meta}, #vacc{resp=Resp0}=Acc) ->
+all_dbs_or_dbs_info_callback({meta, _Meta}, #vacc{resp=Resp0}=Acc) ->
     {ok, Resp1} = chttpd:send_delayed_chunk(Resp0, "["),
     {ok, Acc#vacc{resp=Resp1}};
-all_dbs_callback({row, Row}, #vacc{resp=Resp0}=Acc) ->
+all_dbs_or_dbs_info_callback({row, Row}, #vacc{resp=Resp0}=Acc) ->
     Prepend = couch_mrview_http:prepend_val(Acc),
-    case couch_util:get_value(id, Row) of <<"_design", _/binary>> ->
-        {ok, Acc};
-    DbName ->
-        {ok, Resp1} = chttpd:send_delayed_chunk(Resp0, [Prepend, ?JSON_ENCODE(DbName)]),
-        {ok, Acc#vacc{prepend=",", resp=Resp1}}
+    [PathParts] = Acc#vacc.req#httpd.path_parts,
+    case ?b2l(PathParts) of
```

Comment:
Since the most of the function clause is just one big case statement, might look better to just do the `path_parts` check in the function head and have two function clauses.

Reply:
Changed, thanks for the review.